python/src/puml/helpers/requirements_helpers.py

```diff
@@ -0,0 +1,132 @@
+            requirements.append({
+                "type": type,
+                "title": title,
+                "name": name,
+                "id": id,
+                "shall": shall,
+                "attributes": attributes,
+                "connected": connected,
+            })
+    return requirements
+
+
+def _get_requirement_type(attributes: List[dict]) -> str:
+    """
+    Helper method for extracting the TADI type of a requirement
+
+    Args:
+        attributes (List[dict]): A List of a requirement definitions attributes
+
+    Returns:
+        The value of the TADI attribute
+    """
+    if not attributes:
+        attributes = [{}]
+    for attribute in attributes:
+        if attribute["name"] == "TADI":
+            print(attribute["value"])
+            return attribute["value"]
+
+
+def _get_connected_requirements(req: dict, reqs: dict) -> List[dict]:
+    """
+    Helper method that finds the parent and child requirement definitions for a specified requirement
+
+    Args:
+        req (dict): A requirement defintion
+        reqs (dict): A dictionary containing all requirement definitions
+
+    Returns:
+        A list of parent and child requirement definitions for the specified requirement
+    """
+    connected_reqs = []
+    if not reqs:
+        return connected_reqs
+
+    for struct in reqs:
+        if req != struct:
+            req_id = req.structure["req"]["id"]
+
+            child = _get_child_requirements(req_id, reqs[struct])
+            if child:
+                connected_reqs.append(child)
+
+            parent = _get_parent_requirements(req_id, reqs[struct])
+            if parent:
+                connected_reqs.append(parent)
+    return connected_reqs
+
+
+def _get_child_requirements(req_id: str, other_req: dict) -> Optional[dict]:
+    """
+    Helper method that calls _get_requirement_ancestry() for requirements with children
+
+    Args:
+        req_id: The requirement ID of the requirement for which we are searching for children
+        other_req (dict): The definition for the requirement currently being checked against req_id
+
+    Returns:
+        Optionally returns a dict containing any found child relationships
+    """
+    return _get_requirement_ancestry(req_id, other_req, "children")
+
+
+def _get_parent_requirements(req_id: str, other_req: dict) -> Optional[dict]:
+    """
+    Helper method that calls _get_requirement_ancestry() for requirements with parents
+
+    Args:
+        req_id: The requirement ID of the requirement for which we are searching for parents
+        other_req (dict): The definition for the requirement currently being checked against req_id
+
+    Returns:
+        Optionally returns a dict containing any found parent relationships
+    """
+    return _get_requirement_ancestry(req_id, other_req, "children")
+
+
+def _get_requirement_ancestry(req_id: str, other_req: dict, direction: str) -> Optional[dict]:
```

Comment:
Not sure off hand just yet what is going on with this, but believe this method is the culprit. The proper ID is not being followed it seems. 

I passed
```
(venv) gitpod /workspace/aac-puml/python (10-update-puml-requirements-diagram-template) $ aac puml-requirements ./tests/calc/spec/Add_SRS.yaml ./tests/calc/diagrams/requirements
```

But the diagram I got back was for subtraction
```
@startuml Requirements Diagram

mainframe **req** [Package] **** [Requirements]

object "<<Test>>" as Subtract {
    id = SUB-1
    Text = When receiving a subtract message, the calculator shall respond with the difference between the provided values.
    TADI = Test
    
}

SM-1 +-- SUB-1

SM-1 +-- SUB-1




@enduml
```

Reply:
I believe this was due to the explicit_start=True flag being removed from yaml.dump().  I re-added the flag and it seems to work now.